### DISCUSSION TITLE
Force Android app to use G6 alternate channel?

### DISCUSSION THREAD

**Discussion by schmadde** [NightscoutFoundation/xDrip#1993] [Q&A] [answered]
Hi

is there a way to use a full Android device use the alternate Channel of a G6 Transmitter so I can run in parallel to a phone with xdrip+ (or patched Dexcom app for that matter)?

I would need this for a full android watch or an android based cycle computer. 

I looked at the source code and have a hard time understanding what happens there. It seems to me that using the alternate channel is controlled by the function usingAlt() in Ob1G5StateMachine.java - is this correct? There seems to be a method called WholeHouse.isLive() - what does it do and how is it controlled? Can it be used somehow for this purpose?

**Comment by tolot27** [comment, marked as answer]
I'm working on this feature because the requirement is to get t:slim running side-by-side with xDrip. A PR can be expected soon.

**Reply by jonprogers** [reply]
Hi, I'm currently using the t:slim side by side with xDrip, all has been very smooth. The t:slim connects on the alt/medical device channel and xDrip on the main/mobile device channel.

If I can help test anything please let me know.

This is a UK model t:slim, running the ControlQ firmware (moonlight v7.4) and xDrip on an LG G5, android 8 on the 15th Jan nightly, I've been using this hardware for a couple of years and all previous versions have also worked fine.

Jon.

**Reply by Navid200** [reply]
That's true.  T:slim already works perfectly fine side-by-side with xDrip.

There has been a request to allow xDrip to use the alternative slot itself hoping that would allow another app on the same phone to also connect to the G6.  The concern there is that this may not be possible because both apps would have to use the same BLE link.

I suspect we are misunderstanding what tolot27 means.

**Reply by schmadde** [reply]
Thanks a lot for implementing it. 

I am having trouble to get it to work, though. From the documentation I figured that I need to set this to "1" to use the receiver channel and if not set it would be "2" meaning the default (smartphone) channel?

I tried with my secondary smartphone, a Nokia 8 which was running xdrip+ fine for at least a year. I does seem to connect to the transmitter every 30 minutes or so when setting "1", but never gets a reading. 

On my Android bike computer (for which I wanted this in the first place) the setting is greyed out. Also the "engineering mode" setting is greyed out so I cannot set it. 

Any Ideas how to get this enabled? The device does not have google services on it but that should not prevent setting engineering mode, right?

**Reply by Navid200** [reply]
This explains how you can enable Engineering Mode.
https://navid200.github.io/xDrip/docs/Engineering-Mode.html

**Comment by Navid200** [comment]
The capability has been added:
https://navid200.github.io/xDrip/docs/G6_slot.html

**Comment by loopy321** [comment]
@tolot27 Thanks for this feature!!  Any idea whether the alternate slot can be set for the Wear collector?  

I'm using a SW50, but it appears to want to use slot 2 regardless of the setting.  (discussion question by me here: https://github.com/NightscoutFoundation/xDrip/discussions/2024).
